app/src/main/java/app/revanced/integrations/patches/BackToExitPatch.java

```diff
@@ -0,0 +1,54 @@
+package app.revanced.integrations.patches;
+
+import com.google.android.apps.youtube.app.watchwhile.WatchWhileActivity;
+import app.revanced.integrations.utils.LogHelper;
+
+public class BackToExitPatch {
+    private static long backPressedCount = 0;
+
+    /**
+     * State whether scroll position reaches the top
+     */
+    public static boolean isHomeScrolledTop = false;
+
+    /**
+     * Detect event when back button is pressed
+     *
+     * @param activity is used when closing the app
+     */
+    public static void exitOnBackPressed(WatchWhileActivity activity) {
+        // Check scroll position reaches the top in home feed
+        if (!isHomeScrolledTop) return;
+
+        // Close the current activity once counter reached 1
+        if (backPressedCount <= 0) {
+            LogHelper.printDebug(() -> "BackPressed detected");
+
+            backPressedCount++;
+        } else {
+            LogHelper.printDebug(() -> "Activity is finished");
+
+            activity.finish();
+        }
```

Comment:
This makes no sense. Why is this check required? The activity should be finished when `isHomeScrolledTop` is true immediately, when `exitOnBackPressed` is called.